### DISCUSSION TITLE
Comment syntax for JSDoc

### DISCUSSION THREAD

**Discussion by jaredcwhite** [ruby2js/ruby2js#120] [Ideas]
I've started playing around with JSDoc comments in a library I've built with Ruby2JS and it's nice that they're similar to YARD, and what's cool is I can easily generate TypeScript definitions without actually writing TS.

The only thing that's a little iffy is the syntax you have to use. Essentially my code has to look like this:

```ruby
=begin*
 * This is way cool.
 *
 * @param name {string} Wowee
 * @param selector {string} Whoa
=end
  def self.target(name, selector)
    # etc.
```

In order to get this:

```js
  /**
 * This is way cool.
 *
 * @param name {string} Wowee
 * @param selector {string} Whoa
*/

  static target(name, selector) {
```

I'm thankful that kind of multiline commenting *does* work in the Ruby > JS transpilation, but I'm wondering if there's anything else we can do to make it more seamless. Maybe something like this where it's just JS comment syntax embedded within?

```ruby
  #/**
  # * This is way cool.
  # *
  # * @param name {string} Wowee
  # * @param selector {string} Whoa
  # */
  def self.target(name, selector)
    # etc.
```

Any thoughts @rubys ?

**Comment by rubys** [comment]
My first thoughts is that instead of trying to find a way to essentially escape JSDoc comments in a ruby syntax, perhaps it would be better to create a yard filter that transpiles yard documentation into jsdoc syntax.

**Reply by jaredcwhite** [reply]
Interesting, that might be the way to go!

**Comment by jaredcwhite** [comment]
@rubys I could be wrong, but I think comments don't end up in the AST handled by filters. Seems to be done at the parser level in `converter.rb`. Might have to think of a way to tweak things there specifically…

**Reply by rubys** [reply]
While you are correct that the comments don't go into the AST, they are captured into a hash which is indexed by the AST node that they go with.

So, in practice, if you are in the body of (for example) an `on_def` method in a filter, `@comments[@ast]` would contain the comments associated with that method definition.  You should be able to replace any or all of the comments you find in there with an object of your construction.  That could even be a new type, and `converter.rb` could look for that type and handle it in a unique way.

**Reply by jaredcwhite** [reply]
Oh cool, thanks for the pointer.
